cmd/server/assets/codestatus/show.html

```diff
@@ -18,45 +18,88 @@ <h1>Verification code status</h1>
       Status of a verification code previously shared with your patient.
     </p>
 
-    <div class="card mb-3">
+    <div class="card mb-3" style="max-width: 30rem">
       <div class="card-header">
         Code status
       </div>
-      <div class="card-body">
-        <div class="list-group">
+      <div class="list-group">
+        <div class="list-group-item">
           <h5 class="mb-1">UUID</h5>
           <p class="mb-1 text-monospace">{{.uuid}}</p>
         </div>
-        <div class="list-group">
+        <div class="list-group-item">
           <h5 class="mb-1">Status</h5>
           <p class="mb-1">{{.status}}</p>
         </div>
-        <div class="list-group">
+        <div class="list-group-item">
           <h5 class="mb-1">Expiry</h5>
           <span id="code-expires-at" class="sm text-danger"></span>
         </div>
       </div>
+      <button type="button" id="invalidate" class="btn btn-danger btn-sm">Invalidate code now</button>
     </div>
 
     <div>
-      <p>
-        <a href="/code/status">&larr; Enter another code</a>
-      </p>
+      <a href="/code/status">&larr; Enter another code</a>
     </div>
   </main>
 
   {{template "scripts" .}}
   {{template "codescripts" .}}
 
   <script type="text/javascript">
+    let $buttonInvalidate = $('button#invalidate');
     let $codeExpiresAt;
     let expires = {{.expires }};
+    let uuid = {{.uuid }};
+    let codeCountdown;
 
     $(function() {
       $codeExpiresAt = $('#code-expires-at');
       // Start countdown
-      longCodeCountdown = countdown($codeExpiresAt, expires);
+      codeCountdown = countdown($codeExpiresAt, expires, function() {
+        // Disable the submit if already expired.
+        $buttonInvalidate.prop('disabled', true);
+      });
+
+      $buttonInvalidate.on('click', function() {
+        // Disable the submit button so we only issue one code
+        $buttonInvalidate.prop('disabled', true);
+        let data = {
+          uuid: uuid
+        };
+        invalidateCode(data);
+      });
     });
+
+    function invalidateCode(data) {
```

Comment:
I'm not convinced all this javascript is necessary. Could we just have the button be a regular form/link and use old-school HTML and forms?

Reply:
We could, but I was worried that would be a worse experience with the submit page ultimately redirecting back to here. Without ajax there would be a funny page-refresh (but it does eliminate the timer reset js and such).

Reply:
The Internet is pretty fast, and I think the simplicity it gets us is well worth the UX tradeoff.

Reply:
agree the form is simpler to maintain - if you want to check this in and change it later, up to ou.

Reply:
/hold

Nah let's hold this. I'll look at a form.
Pulled out the API: https://github.com/google/exposure-notifications-verification-server/pull/380 and I'll focus on the UX separately